src/backends/gr.jl

```diff
@@ -1814,17 +1814,36 @@ function gr_add_title(sp, viewport_plotarea, viewport_subplot)
         gr_set_font(titlefont(sp), sp)
         loc = sp[:titlelocation]
         if loc === :left
-            xpos = viewport_plotarea[1]
-            halign = GR.TEXT_HALIGN_LEFT
+            xpos, ypos = viewport_plotarea[1], viewport_subplot[4]
+            halign, valign = GR.TEXT_HALIGN_LEFT, GR.TEXT_VALIGN_TOP
+        elseif loc === :center
+            xpos, ypos = +(viewport_plotarea[1:2]...) / 2, viewport_subplot[4]
+            halign, valign = GR.TEXT_HALIGN_CENTER, GR.TEXT_VALIGN_TOP
         elseif loc === :right
-            xpos = viewport_plotarea[2]
-            halign = GR.TEXT_HALIGN_RIGHT
+            xpos, ypos = viewport_plotarea[2], viewport_subplot[4]
+            halign, valign = GR.TEXT_HALIGN_RIGHT, GR.TEXT_VALIGN_TOP
         else
-            xpos = gr_view_xcenter(viewport_plotarea)
-            halign = GR.TEXT_HALIGN_CENTER
+            xpos = gr_view_xposition(viewport_plotarea, loc[1])
+            ypos = gr_view_yposition(viewport_plotarea, loc[2])
+            halign = sp[:titlefonthalign]
+            valign = sp[:titlefontvalign]
+            if halign == :left
+                halign = GR.TEXT_HALIGN_LEFT
+            elseif halign == :hcenter || halign == :center
+                halign = GR.TEXT_HALIGN_CENTER
+            elseif halign == :right
+                halign = GR.TEXT_HALIGN_RIGHT
+            end
+            if valign == :top
+                valign = GR.TEXT_VALIGN_TOP
+            elseif valign == :vcenter || valign == :center
+                valign = GR.TEXT_VALIGN_HALF
+            elseif valign == :bottom
+                valign = GR.TEXT_VALIGN_BOTTOM
+            end
```

Comment:
```suggestion
            halign = gr_halign(sp[:titlefonthalign])
            valign = gr_valign(sp[:titlefontvalign])
```

If think the helpers `gr_halign` and `gr_valign` defined in this file will work no ?

Reply:
Yeah, good call, thanks